tests/exchanges/Ethfinex.test.js

```diff
@@ -0,0 +1,234 @@
+          })
+        })
+
+        test('when the total amount is greater than all bids', () => {
+          const getAvailableMarkets = jest.spyOn(exchange, 'getAvailableMarkets')
+          getAvailableMarkets.mockResolvedValue(markets)
+
+          const _getRawOrderBook = jest.spyOn(exchange, '_getRawOrderBook')
+          _getRawOrderBook.mockResolvedValue({
+            bids: [{ price: '1.00', amount: '200' }, { price: '2.00', amount: '100' }],
+            asks,
+          })
+
+          expect(exchange.computePrice(symbol, amount, true, fee)).resolves.toMatchObject({
+            Ethfinex: {
+              exchangeName: 'Ethfinex',
+              tokenAmount: amount,
+              tokenSymbol: symbol,
+              timestamp: expect.any(Number),
+              error: noLiquidityMessage,
+            },
+          })
+        })
+      })
+
+      describe('when buying', () => {
+        test('when there are no asks', () => {
+          const getAvailableMarkets = jest.spyOn(exchange, 'getAvailableMarkets')
+          getAvailableMarkets.mockResolvedValue(markets)
+
+          const _getRawOrderBook = jest.spyOn(exchange, '_getRawOrderBook')
+          _getRawOrderBook.mockResolvedValue({ asks: [], bids })
+
+          expect(exchange.computePrice(symbol, amount, false, fee)).resolves.toMatchObject({
+            Ethfinex: {
+              exchangeName: 'Ethfinex',
+              tokenAmount: amount,
+              tokenSymbol: symbol,
+              timestamp: expect.any(Number),
+              error: noLiquidityMessage,
+            },
+          })
+        })
+
+        test('when the total amount is greater than all asks', () => {
+          const getAvailableMarkets = jest.spyOn(exchange, 'getAvailableMarkets')
+          getAvailableMarkets.mockResolvedValue(markets)
+
+          const _getRawOrderBook = jest.spyOn(exchange, '_getRawOrderBook')
+          _getRawOrderBook.mockResolvedValue({
+            bids,
+            asks: [{ price: '3.00', amount: '100' }, { price: '4.00', amount: '200' }],
+          })
+
+          expect(exchange.computePrice(symbol, amount, false, fee)).resolves.toMatchObject({
+            Ethfinex: {
+              exchangeName: 'Ethfinex',
+              tokenAmount: amount,
+              tokenSymbol: symbol,
+              timestamp: expect.any(Number),
+              error: noLiquidityMessage,
+            },
+          })
+        })
+      })
+    })
+
+    test('returns the average and total prices for a buy order across price levels', () => {
+      const getAvailableMarkets = jest.spyOn(exchange, 'getAvailableMarkets')
+      getAvailableMarkets.mockResolvedValue(markets)
+
+      const _getRawOrderBook = jest.spyOn(exchange, '_getRawOrderBook')
+      _getRawOrderBook.mockResolvedValue(orderBook)
+
+      expect(exchange.computePrice(symbol, amount, false, fee)).resolves.toMatchObject({
+        Ethfinex: {
+          exchangeName: 'Ethfinex',
+          totalPrice: 1800.0,
```

Comment:
Worth putting in the calculation here and on the other two happy-path tests below?

```
// (3 * (2/5)) + (4 * (3/5))
```

Reply:
yeah I would define this as a local variable so it's clear how the number is computed

Reply:
@grahamperich added! Also set a minimum threshold for function coverage based on the current tests: e44e382